x/twap/api.go

```diff
@@ -41,12 +41,47 @@ func (k Keeper) GetArithmeticTwap(
 	quoteAssetDenom string,
 	startTime time.Time,
 	endTime time.Time,
+) (sdk.Dec, error) {
+	arithmeticStrategy := &arithmetic{k}
+	return k.getTwap(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, endTime, arithmeticStrategy)
+}
+
+// GetArithmeticTwapToNow returns GetArithmeticTwap on the input, with endTime being fixed to ctx.BlockTime()
+// This function does not mutate records.
+func (k Keeper) GetArithmeticTwapToNow(
+	ctx sdk.Context,
+	poolId uint64,
+	baseAssetDenom string,
+	quoteAssetDenom string,
+	startTime time.Time,
+) (sdk.Dec, error) {
+	arithmeticStrategy := &arithmetic{k}
+	return k.getTwapToNow(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, arithmeticStrategy)
+}
+
+// GetBeginBlockAccumulatorRecord returns a TwapRecord struct corresponding to the state of pool `poolId`
+// as of the beginning of the block this is called on.
+// This uses the state of the beginning of the block, as if there were swaps since the block has started,
+// these swaps have had no time to be arbitraged back.
+// This accumulator can be stored, to compute wider ranged twaps.
+func (k Keeper) GetBeginBlockAccumulatorRecord(ctx sdk.Context, poolId uint64, asset0Denom string, asset1Denom string) (types.TwapRecord, error) {
+	return k.getMostRecentRecord(ctx, poolId, asset0Denom, asset1Denom)
+}
+
+func (k Keeper) getTwap(
```

Comment:
let's add godoc please explaining that this is a more generalized method taking a strategy. Based on the strategy taken, it would execute different logic